### PR TITLE
[#2064] Introduce PendingPubAcks, remove CommandSubscriptionsManager

### PR DESCRIPTION
**Pull request by calohmn**
This is for #2064:
Refactoring to make handling of PUBACKs in the MQTT adapter independent of the command handling.

After having extracted the `PendingPubAcks` class out of `CommandSubscriptionsManager`, there was no real need for a separate  `CommandSubscriptionsManager` class anymore (using just a `Map` in `MqttDeviceEndpoint` instead).

### CODE REVIEW

File: adapters/mqtt-vertx-base/src/main/java/org/eclipse/hono/adapter/mqtt/PendingPubAcks.java [hunk 1]
```diff
@@ -0,0 +1,139 @@
+/*******************************************************************************
+ * Copyright (c) 2016, 2021 Contributors to the Eclipse Foundation
```
[the remark this answers is not in the document]
Comment (calohmn) [reply adapters/mqtt-vertx-base/src/main/java/org/eclipse/hono/adapter/mqtt/PendingPubAcks.java:2]: You could see it as a renamed `CommandSubscriptionsManager` class, that's why I kept its header.

[the remark this answers is not in the document]
Comment (kaniyan) [reply adapters/mqtt-vertx-base/src/main/java/org/eclipse/hono/adapter/mqtt/PendingPubAcks.java:2]: It didn't show that it has been renamed rather `CommandSubscriptionsManager` as deleted and `PendingPubAcks` as a new file added. If it is renaming, the it makes sense also to keep `2016`.

File: adapters/mqtt-vertx-base/src/main/java/org/eclipse/hono/adapter/mqtt/AbstractSubscription.java [hunk 2]
```diff
@@ -165,4 +166,62 @@ public final void logSubscribeFailure(final Span span, final Throwable error) {
...
+        items.put(Fields.EVENT, "removing subscription");
+        items.put(LOG_FIELD_TOPIC_FILTER, getTopic());
+        span.log(items);
+    }
+
+    /**
+     * The key to identify a subscription.
+     * To be used for a kind of subscription where there can only be one subscription for a given device.
+     */
+    static final class DefaultKey implements Key {
```
[the remark this answers is not in the document]
Comment (calohmn) [reply adapters/mqtt-vertx-base/src/main/java/org/eclipse/hono/adapter/mqtt/AbstractSubscription.java:183]: The `DeviceKey` class is in the `org.eclipse.hono.deviceregistry.service.device` package in the `device-registry-base` module and it's documented as to be used for the Device Registry Management API. I think that class should better be kept there for its defined purpose.
For the MQTT subscription key I would also like to have the possibility for future extensions (like include an endpoint string in the subscription key), so I would rather use a specific MQTT subscription key interface and implementation class.

**Review by kaniyan — APPROVED** [review]
LGTM

**Pull request merged**